src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = (',')) {
+    let result = '';
+
+    if (!this.length) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        result += nullCheck(this[i]);
+      } else {
+        result += nullCheck(this[i]) + separator;
+      }
```

Comment:
It looks more readable I suppose:

```suggestion
      result += nullCheck(this[i]);
      
      if (i !== this.length - 1) {
        result += separator;
      }
```

Reply:
Agree, fixed